File: src/dotnet/Ansight.Core/Pairing/Models/PairingTlsPin.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Time-bounded TLS SubjectPublicKeyInfo pin authorized by a protocol-v2 config.
/// </summary>
public sealed class PairingTlsPin
{
    public required string TlsSpkiSha256 { get; set; }

    public required string NotBefore { get; set; }

    public required string NotAfter { get; set; }
}

File: src/ios/Sources/AnsightCore/PairingTlsPin.swift
import Foundation

public struct PairingTlsPin: Sendable, Codable, Equatable {
    public var tlsSpkiSha256: String
    public var notBefore: String
    public var notAfter: String

    public init(tlsSpkiSha256: String, notBefore: String, notAfter: String) {
        self.tlsSpkiSha256 = tlsSpkiSha256
        self.notBefore = notBefore
        self.notAfter = notAfter
    }
}
